code/game/objects/items/weapons/glowstick.dm

```diff
@@ -52,3 +52,57 @@
 #undef GLOW_GREEN
 #undef GLOW_RED
 #undef GLOW_BLUE
+
+/obj/item/clothing/accessory/glowstick //Maybe later convert the rest over
+	name = "glowstick"
+	desc = "A glowstick filled with luminescent liquid. It has a string on it so you can wear it."
+	icon_state = "glowstick"
+	_color = "glowstick" //for setting accessory states
+	color = rgb(255, 255, 255) //for coloring the accessory
+
+/obj/item/clothing/accessory/glowstick/attack_self(mob/user)
+	to_chat(user, "<span class='notice'>You crack \the [src] and it lights up.")
+	set_light(2, l_color = color)
+
+/obj/item/clothing/accessory/glowstick/suicide_act(mob/user)
+	user.visible_message("<span class='danger'>[user] is breaking open \the [src.name] and eating the liquid inside! It looks like \he's  trying to commit suicide!</span>")
```

Comment:
Yeah but it doesn't break the item when they suicide.

Also, `\the [src]` instead of `src.name`, and you have a duplicate space after `\he's`